### ISSUE TITLE
[NEW FEATURE] height resizeable panels

### ISSUE THREAD

**Issue by jolsen-mitsu** [mrjackwills/oxker#50]
Right now the relative size of the panes is not user changeable. 

It would be nice if there were key commands, setting file settings, and interactive mouse support to change the size of the panes, most notably the height.

For example, I have a lot of containers and I do not often use the logs pane. Even if I size the window really big, only a portion of my containers are visible, and the majority of the window is taken up by the logs section which is blank and wasted space in this scenario.  I would love to be able to shrink the number of rows for the logs pane and increase the number of rows for the container pane.  Possibly also shrink the height of the cpu/memory/ports section so I can give the containers pane the bulk of the window.

The panes would retain their relative sizes when the window is resized and may need to temporarily re-adjust pane size ratios if the user makes the window size very small.

It would be nice if this change can be done via shortcut keys, but also via mouse by clicking and dragging the pane frame (tmux supports this and may have implementation/code ideas to lean on)

It would be very nice if the changes were remembered between launches.

**Comment by creechy** [comment]
And for the logs pane in particular, the ability to toggle it off completely. Maybe the other panes too, but opening logs on demand would be most useful.

**Comment by mrjackwills** [comment]
Interesting, hadn’t considered that use case. It would be easy to add a section in the [config.toml](https://github.com/mrjackwills/oxker/blob/main/example_config/example.config.toml) to set the heights, or visibilities, of the sections.

But a feature to change these whilst running `oxker` would be much better, need some time to think about the best way to achieve it.

**Comment by mrjackwills** [comment]
@creechy @jolsen-mitsu Would you be able to test this feature for me?

I can push a branch, which you need to fetch and build yourself, or I can build a binary and upload here (providing to tell me which OS and architecture you would require)

**Comment by creechy** [comment]
@mrjackwills I probably can next week. I had claude write me a quick app, but its been challenging to get it to perform well. Ha! anyways on macOS/Apple Silicon.

**Comment by mrjackwills** [comment]
@creechy Thanks, there should be a binary attached to this comment.

You can resize the Logs panel by using either the `-` or `=` key. Also, if you use a `config.toml/json` configuration file, you can now add `show_logs=false`, which will hide the logs - the logs can then be re-shown again using the `=` key

[oxker_aarch64_apple_darwin.zip](https://github.com/user-attachments/files/20738474/oxker.zip)

**Comment by mrjackwills** [comment]
Good idea, maybe the `Delete` key to quickly toggle the logs panel - although if I remember correctly theres an issue with this key on MacOs v Windows, so maybe `\` instead?

**Comment by creechy** [comment]
I'd go with something other than `Delete`. `\` works or even just `l` (ell).

**Comment by mrjackwills** [comment]
@creechy Thanks, there's another binary attach to this comment, which *should* now toggle the logs section with the `\` key, whilst also saving the height of the logs panel as suggested.

[oxker.zip](https://github.com/user-attachments/files/20739402/oxker.zip)

**Comment by mrjackwills** [comment]
@creechy There should be another binary attached to this comment.

It *should* toggle the logs panel with the `\` key, all whilst saving the height of said logs panel. The `\` key is configurable in the keymap as `log_section_toggle`

[oxker_aarch64-apple-darwin.zip](https://github.com/user-attachments/files/20739507/oxker_aarch64-apple-darwin.zip)

**Comment by mrjackwills** [comment]
Yeah I came across that bug, I think it's now fixed, I'll post another binary at some point, but I've just found a potential huge performance increase so it may take a few days

**Comment by creechy** [comment]
I feel like the later would be better, then. no hidden functionality. `\` to only toggle, `-` & `=` only resize when visible.

**Comment by marlop352** [comment]
Having a way to configure the height of the logs panel would be really good, being able to scroll it horizontally too.
